plugin/src/main/java/org/jenkinsci/plugins/casc/ConfigurationAsCode.java

```diff
@@ -568,9 +568,12 @@ private void listElements(Set<Object> elements, Set<Attribute> attributes) {
                 .map(Attribute::getType)
                 .map(Configurator::lookup)
                 .filter(Objects::nonNull)
+                .map(Configurator::getConfigurators)
+                .flatMap(Collection::stream)
                 .forEach(configurator -> {
-                    elements.addAll(configurator.getConfigurators());
-                    listElements(elements, configurator.describe());
+                    if (elements.add(configurator)) {
+                        listElements(elements, ((Configurator<?>) configurator).describe());
```

Comment:
cast is required for some weird generic erasure as getConfigurator return `Set<Configurator>` and not `Set<Configurator<?>>`, which break elsewhere.
Java generics sucks.